tests/pysharedutils/dicts_tests.py

```diff
@@ -281,3 +281,100 @@ def test_get_dict_properties_without_strict_and_invalid_property(self):
             'comment.user.username'
         )
         assert_equal(output, {'comment.user.username': None})
+
+
+class TestMapDictKeys:
+
+    def test_map_dict_keys(self):
+        obj = {
+            'first_name': 'Joe',
+            'middle_name': 'P',
+            'last_name': 'Smith'
+        }
+        map_obj = {
+            'first_name': 'given_name',
+            'last_name': 'surname',
+            'mis_spelled_key': 'mis_spelled_value'
+        }
+        output = pysharedutils.map_dict_keys(obj, map_obj)
+        expected_output = {
+            'given_name': 'Joe',
+            'middle_name': 'P',
+            'surname': 'Smith'
+        }
+        assert_equal(output, expected_output)
+
+    def test_map_dict_keys_with_one_level(self):
+        obj = {
+            'first_name': 'Joe',
+            'user': {
+                'username': 'joe@example.com'
+            }
+        }
+        map_obj = {
+            'first_name': 'given_name',
+            'user.username': 'email',
```

Comment:
This is interesting behavior - it took me a minute to understand it. I thought the output would be: `{'given_name': '...', 'email': '...'}`

Some other options that may make a bit more sense:

1. `{'user.username': 'user.email'}`
2. `{'user': {'username': 'email'}}`